main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -805,6 +805,10 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       }
     }
 
+    def NewObject: Rule1[ParsedAst.Expression] = rule {
+      SP ~ keyword("object") ~ WS ~ atomic("##") ~ Names.JavaName ~ SP ~ WS ~ "{" ~ optWS ~ "}" ~> ParsedAst.Expression.NewObject
```

Comment:
Two things:

- You insist on hard whitespace before the first brace? We could get away with optional whitespace.
- The second SP should be after the last brace (thus the two SPs span the entire range of the object expression).